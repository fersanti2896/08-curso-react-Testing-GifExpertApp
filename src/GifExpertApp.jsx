import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
    const [ categories, setCategories ] = useState([ 'Shrek' ]);
    
    /**
     * Función que agrega una categoría al arreglo de categorías.
     * @param {string} newCategory Ejemplo: Shrek
     */
    const onAddCategory = ( newCategory ) => {
        const cleanCategory = newCategory[0].toUpperCase() + newCategory.substring(1).toLowerCase();

        if (categories.includes(cleanCategory)) return;

        setCategories([ cleanCategory, ...categories ]);
    }

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory newCategory={ onAddCategory }/>

            { categories.map( cat => ( <GifGrid key={ cat } category={ cat } /> ) )}
            
        </>
    )
}
