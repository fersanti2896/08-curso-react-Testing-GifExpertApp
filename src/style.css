
html {
  background-color: #21232a;
  color: white;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 1.3rem;
}

body {
  padding: 60px;
}

input {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgb(97, 32, 158);
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 100%;
}

.card {
  align-content: center;
  background-color: #8e92a0c2;
  border: none;
  box-shadow: 2px 2px 6px 0px rgba(0, 0, 0, 0.3);
  border-radius: 20px;
  margin-bottom: 10px;
  margin-right: 10px;
  overflow: hidden;
  width: 250px;
  transition: transform .1s;
}

.card:hover {
  transform: translateY(5px);
  box-shadow: 2px 2px 6px 0px rgba(0, 0, 0, 0.3);
}

.card p {
  /* background-color: #83757b; */
  padding: 0 20px 20px;
  text-align: center;
  text-transform: uppercase;
}

.card img {
  width: 130%;
  border-top-left-radius: 20px;
  border-top-right-radius: 20px;
}

.card-grid{
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

@media (max-width: 768px) {
  .card {
    grid-template-columns: repeat(3, 1fr);
  }
}

@media (max-width: 500px) {
  .card {
    grid-template-columns: repeat(2, 1fr);
  }

  .card p {
    text-align: center;
  }
}