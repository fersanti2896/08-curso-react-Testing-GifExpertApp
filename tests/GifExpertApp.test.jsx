import { render, screen } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {
    test('Debe hacer match con el snapshot.', () => { 
        const { container } = render( <GifExpertApp /> );

        expect( container ).toMatchSnapshot();
    });

    test('Debe mostrar en un h1 GifExpertApp como tÃ­tulo.', () => {
        render( <GifExpertApp /> );

        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toBe('GifExpertApp');
    });
});