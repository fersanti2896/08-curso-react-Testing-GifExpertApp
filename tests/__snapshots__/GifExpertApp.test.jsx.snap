// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe hacer match con el snapshot. 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <h5 />
  <h5>
    Cargando...
  </h5>
  <div
    class="card-grid"
  />
</div>
`;
